Tests failing on release builds

After pulling the latest changes compiling a release build fails with segfaults during tests. Compiling a debug build works.

~~~~
./compile.sh -r
[...]
-- Using CMake version 3.22.0
-- Building with XDG support
-- Found the following ICU libraries:
--   uc (required)
--   dt (required)
-- GCC >6.2.0 breaks -03, setting -02
-- Not using ThreadSanitizer.
-- Not using AddressSanitizer.
-- Adding linker flags for GLVND.
-- Enabling link-time optimizations
-- Version of Widelands Build is 1.1~git25528[776dcf0@master](Release)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/kaputtnik/Quellcode/widelands-repo/widelands/build
[...]
[100%] Built target wl_tests
Test project /home/kaputtnik/Quellcode/widelands-repo/widelands/build
    Start 1: test_ai
1/6 Test #1: test_ai ..........................***Exception: SegFault  0.23 sec

    Start 2: test_base
2/6 Test #2: test_base ........................***Exception: SegFault  0.15 sec

    Start 3: test_economy
3/6 Test #3: test_economy .....................***Exception: SegFault  0.87 sec

    Start 4: test_io_filesystem
4/6 Test #4: test_io_filesystem ...............***Exception: SegFault  0.14 sec

    Start 5: notifications_test
5/6 Test #5: notifications_test ...............   Passed    0.01 sec
    Start 6: test_scripting
6/6 Test #6: test_scripting ...................***Exception: SegFault  0.15 sec


17% tests passed, 5 tests failed out of 6

Total Test time (real) =   1.56 sec

The following tests FAILED:
          1 - test_ai (SEGFAULT)
          2 - test_base (SEGFAULT)
          3 - test_economy (SEGFAULT)
          4 - test_io_filesystem (SEGFAULT)
          6 - test_scripting (SEGFAULT)
Errors while running CTest
make[2]: *** [CMakeFiles/_run_all_tests.dir/build.make:70: CMakeFiles/_run_all_tests] Fehler 8
make[1]: *** [CMakeFiles/Makefile2:2675: CMakeFiles/_run_all_tests.dir/all] Fehler 2
make: *** [Makefile:146: all] Fehler 2
~~~~

This happens also with a clean build (deleting the build directory prior compiling).

Running regression tests gives no errors (after compiling with `./compile-sh -r -s`) .

 - OS: archlinux
 - Widelands Version: Build is 1.1~git25528[776dcf0@master](Release)

Some version infos:
- gcc 11.1.0-1 
- cmake 3.22.0-1
- boost 1.76.0-1
- icu 69.1-1
- kernel: 5.15.3-arch1-1

--- a/src/base/test.cc
+++ b/src/base/test.cc
@@ -17,4 +17,11 @@
  *
  */
 
-// Dummy file as cmake cannot handle header only libraries.
+#include "base/test.h"
+
+namespace WLTestsuite {
+Testsuite& all_testsuites() {
+	static Testsuite* t = new Testsuite;
+	return *t;
+}
+}  // namespace WLTestsuite

--- a/src/base/test.h
+++ b/src/base/test.h
@@ -80,7 +80,7 @@
 namespace WLTestsuite {
 using Testcases = std::map<std::string, void (*)()>;
 using Testsuite = std::map<std::string, Testcases>;
-extern Testsuite all_testsuites_;
+Testsuite& all_testsuites();
 
 struct InternalTestcaseInserter {
 	explicit InternalTestcaseInserter(Testcases& tc, const std::string& name, void (*fn)()) {
@@ -158,13 +158,13 @@ inline void do_check_equal(const char* f, uint32_t l, const T1& a, const T2& b) 
 
 #define TEST_EXECUTABLE(name)                                                                      \
 	namespace WLTestsuite {                                                                         \
-	Testsuite all_testsuites_;                                                                      \
 	namespace WLTestsuite_##name {                                                                  \
 		static int main() {                                                                          \
 			bool errors = false;                                                                      \
-			for (const auto& suite : all_testsuites_) {                                               \
+			for (const auto& suite : all_testsuites()) {                                              \
 				for (const auto& test : suite.second) {                                                \
 					try {                                                                               \
+						std::cout << "Running " << suite.first << "::" << test.first << std::endl;       \
 						test.second();                                                                   \
 					} catch (const std::exception& e) {                                                 \
 						errors = true;                                                                   \
@@ -184,7 +184,7 @@ inline void do_check_equal(const char* f, uint32_t l, const T1& a, const T2& b) 
 #define TESTSUITE_START(name)                                                                      \
 	namespace WLTestsuite {                                                                         \
 	namespace WLTestsuite_##name {                                                                  \
-		static Testcases& all_testcases_ = all_testsuites_[#name];
+		static Testcases& all_testcases_ = all_testsuites()[#name];
 
 #define TESTCASE(name)                                                                             \
 	static void testcase_##name();                                                                  \

